Sources/BraintreeCore/RepeatingTimer.swift

```diff
@@ -0,0 +1,70 @@
+import Foundation
+
+final class RepeatingTimer {
+    
+    /// Amount of time, in seconds.
+    private let timeInterval: Int
+    
+    init(timeInterval: Int) {
+        self.timeInterval = timeInterval
+    }
+    
+    private lazy var timer: DispatchSourceTimer = {
+        let timerSource = DispatchSource.makeTimerSource(queue: DispatchQueue.main)
+        timerSource.schedule(
+            deadline: .now() + .seconds(timeInterval),
+            repeating: .seconds(timeInterval),
+            leeway: .seconds(1)
+        )
+        timerSource.setEventHandler { [weak self] in
+            self?.eventHandler?()
+        }
+        return timerSource
+    }()
+    
+    var eventHandler: (() -> Void)?
+    
+    private enum State {
```

Comment:
Another nitpick/styling - to try to use pragma marks according to our internal style guide. I'll grab a link and send this to you internally

![Screenshot 2024-06-26 at 8 17 02 AM](https://github.com/braintree/braintree_ios/assets/35243507/c50e63ff-1988-4c00-8d1c-e200a97f7e54)